react/features/lobby/actions.web.js

```diff
@@ -0,0 +1,209 @@
+// @flow
+
+import { type Dispatch } from 'redux';
+
+import { appNavigate, maybeRedirectToWelcomePage } from '../app';
+import { conferenceLeft, conferenceWillJoin, getCurrentConference, setPassword } from '../base/conference';
+import { hideDialog, openDialog } from '../base/dialog';
+import { getLocalParticipant } from '../base/participants';
+
+import {
+    KNOCKING_PARTICIPANT_ARRIVED_OR_UPDATED,
+    KNOCKING_PARTICIPANT_LEFT,
+    SET_KNOCKING_STATE,
+    SET_LOBBY_MODE_ENABLED,
+    SET_PASSWORD_JOIN_FAILED
+} from './actionTypes';
+import { LobbyScreen } from './components';
+
+declare var APP: Object;
+
+/**
+ * Cancels the ongoing knocking and abandones the join flow.
+ *
+ * @returns {Function}
+ */
+export function cancelKnocking() {
+    return async (dispatch: Dispatch<any>, getState: Function) => {
+        if (typeof APP !== 'undefined') {
+            // when we are redirecting the library should handle any
+            // unload and clean of the connection.
+            APP.API.notifyReadyToClose();
+            dispatch(maybeRedirectToWelcomePage());
+
+            return;
+        }
+
+        dispatch(conferenceLeft(getCurrentConference(getState)));
+        dispatch(appNavigate(undefined));
+    };
+}
+
+/**
+ * Action to hide the lobby screen.
+ *
+ * @returns {hideDialog}
+ */
+export function hideLobbyScreen() {
+    return hideDialog(LobbyScreen);
+}
+
+/**
+ * Tries to join with a preset password.
+ *
+ * @param {string} password - The password to join with.
+ * @returns {Function}
+ */
+export function joinWithPassword(password: string) {
+    return async (dispatch: Dispatch<any>, getState: Function) => {
```

Comment:
Async actions are supposed to return a Promise, but since I didn't have a return value here, I chose to return with an async function that implicitly returns a Promise. We don't do this well in many parts of the app, but thanks to the lenient nature of Javascript it doesn't usually fail.